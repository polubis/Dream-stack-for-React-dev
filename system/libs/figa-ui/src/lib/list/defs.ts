import type { ReactNode } from 'react';

interface ListProps {
  className?: string;
  children: ReactNode;
  ordered?: boolean;
}

interface ListItemProps {
  className?: string;
  children: ReactNode;
}

export type { ListProps, ListItemProps };
