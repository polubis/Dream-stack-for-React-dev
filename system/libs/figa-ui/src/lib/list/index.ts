export * from './list';
export * from './defs';
