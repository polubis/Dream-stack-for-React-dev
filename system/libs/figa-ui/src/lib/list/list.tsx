import type { ListItemProps, ListProps } from './defs';
import c from 'classnames';

const ListItem = ({ className, children }: ListItemProps) => {
  return <li className={className}>{children}</li>;
};

const List = ({ className, children, ordered }: ListProps) => {
  const classes = c('list', className);

  if (ordered) {
    return <ol className={classes}>{children}</ol>;
  }

  return <ul className={classes}>{children}</ul>;
};

export { List, ListItem };
