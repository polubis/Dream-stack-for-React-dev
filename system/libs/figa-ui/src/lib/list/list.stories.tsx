import type { Story, Meta } from '@storybook/react';

import { List, ListItem } from './list';
import { Box } from '../box';

export default {
  component: List,
  title: 'List',
} as Meta;

const Template: Story = () => {
  return (
    <Box spacing={[300]}>
      <List>
        <ListItem>First item</ListItem>
        <ListItem>Second item</ListItem>
        <ListItem>Third item</ListItem>
      </List>

      <List ordered>
        <ListItem>First item</ListItem>
        <ListItem>Second item</ListItem>
        <ListItem>Third item</ListItem>
      </List>
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {};
