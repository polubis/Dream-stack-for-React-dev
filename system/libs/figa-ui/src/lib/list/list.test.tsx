import { render } from '@testing-library/react';
import { List, ListItem } from './list';
import type { ListProps } from './defs';
import type { ReactNode } from 'react';

describe('List can be used when', () => {
  const listFixture = (props: Partial<ListProps> = {}) => {
    const children: ReactNode = (
      <>
        <ListItem>First item</ListItem>
        <ListItem>Second item</ListItem>
        <ListItem>Third item</ListItem>
        <ListItem>Fourth item</ListItem>
      </>
    );

    const result = render(<List children={children} {...props} />);

    return result;
  };

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = listFixture({
      className: 'my-class',
      children: (
        <>
          <ListItem className="first-item-class">First item</ListItem>
          <ListItem>Second item</ListItem>
        </>
      ),
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] allows to render unordered list', () => {
    const { asFragment } = listFixture();

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] allows to render ordered list', () => {
    const { asFragment } = listFixture({ ordered: true });

    expect(asFragment()).toMatchSnapshot();
  });
});
