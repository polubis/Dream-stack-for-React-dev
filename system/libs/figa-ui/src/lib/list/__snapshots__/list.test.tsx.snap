// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List can be used when [FRAGILE] allows to render ordered list 1`] = `
<DocumentFragment>
  <ol
    class="list ordered"
  >
    <li>
      First item
    </li>
    <li>
      Second item
    </li>
    <li>
      Third item
    </li>
    <li>
      Fourth item
    </li>
  </ol>
</DocumentFragment>
`;

exports[`List can be used when [FRAGILE] allows to render unordered list 1`] = `
<DocumentFragment>
  <ul
    class="list unordered"
  >
    <li>
      First item
    </li>
    <li>
      Second item
    </li>
    <li>
      Third item
    </li>
    <li>
      Fourth item
    </li>
  </ul>
</DocumentFragment>
`;

exports[`List can be used when [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <ul
    class="list my-class unordered"
  >
    <li
      class="first-item-class"
    >
      First item
    </li>
    <li>
      Second item
    </li>
  </ul>
</DocumentFragment>
`;
